fdbcli/TenantCommands.actor.cpp

```diff
@@ -228,30 +229,58 @@ ACTOR Future<bool> getTenantCommandActor(Reference<IDatabase> db, std::vector<St
 
 			json_spirit::mValue jsonObject;
 			json_spirit::read_string(tenant.get().toString(), jsonObject);
-			JSONDoc doc(jsonObject);
 
-			int64_t id;
-			std::string prefix;
-			doc.get("id", id);
-			doc.get("prefix", prefix);
+			if (useJson) {
+				json_spirit::mObject resultObj;
+				resultObj["tenant"] = jsonObject;
+				resultObj["type"] = "success";
+				printf("%s\n",
+				       json_spirit::write_string(json_spirit::mValue(resultObj), json_spirit::pretty_print).c_str());
+			} else {
+				JSONDoc doc(jsonObject);
+
+				int64_t id;
+				std::string prefix;
+				doc.get("id", id);
+				doc.get("prefix", prefix);
+
+				printf("  id: %" PRId64 "\n", id);
+				printf("  prefix: %s\n", printable(prefix).c_str());
+			}
 
-			printf("  id: %" PRId64 "\n", id);
-			printf("  prefix: %s\n", printable(prefix).c_str());
 			return true;
 		} catch (Error& e) {
-			state Error err(e);
-			if (e.code() == error_code_special_keys_api_failure) {
-				std::string errorMsgStr = wait(fdb_cli::getSpecialKeysFailureErrorMessage(tr));
-				fprintf(stderr, "ERROR: %s\n", errorMsgStr.c_str());
-				return false;
+			try {
+				wait(safeThreadFutureToFuture(tr->onError(e)));
+			} catch (Error& finalErr) {
+				state std::string errorStr;
+				if (finalErr.code() == error_code_special_keys_api_failure) {
+					std::string str = wait(getSpecialKeysFailureErrorMessage(tr));
+					errorStr = str;
+				} else if (useJson) {
+					errorStr = finalErr.what();
+				} else {
+					throw finalErr;
+				}
+
+				if (useJson) {
+					json_spirit::mObject resultObj;
+					resultObj["type"] = "error";
+					resultObj["error"] = errorStr;
```

Comment:
I don't know how this json library fills out the fields, but this is being assigned with type first and error second, whereas the example in the documentation has error first and type second.

Reply:
The fields get printed in alphabetical order, so the result should match what is documented. Even so, I believe the JSON format treats fields of an object as unordered, so no one should be relying on the order anyway.